pkg/registry/core/pod/storage/storage.go

```diff
@@ -225,7 +223,7 @@ func (r *BindingREST) setPodHostAndAnnotations(ctx context.Context, podUID types
 			return nil, fmt.Errorf("pod %v is already assigned to node %q", pod.Name, pod.Spec.NodeName)
 		}
 		// Reject binding to a scheduling un-ready Pod.
-		if utilfeature.DefaultFeatureGate.Enabled(features.PodSchedulingReadiness) && len(pod.Spec.SchedulingGates) != 0 {
```

Comment:
When calling `storage.Create()`, and if the feature gate is disabled, `schedulingGates` will get dropped, and hence this change cannot be easily covered in UT.

But I manually tested this change by crafting an upgrade/rollback flow (so that we can force binding to a pod persisted with schedulingGates):

- spin up a cluster, with `PodSchedulingReadiness=true`
- create a pod carrying one scheduling gate
- the pod stays in `SchedulingGated` state
- re-start API Server and scheduler, but with `PodSchedulingReadiness=false`
- (without this PR) the pod enters `Running` state (suppose it requests little resource) while also having non-empty `.spec.schedulingGates` present
- (with this PR) setting nodeName for the pod (in scheduler's binding phase) is rejected by API Server, and hence it stays in `Pending` state

Reply:
you can disable the feature gate in the middle of a unit test to drive that same interaction with a call to `defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.PodSchedulingReadiness, false)()`

Reply:
Thanks. I added a flag `flipFeatureGateBeforeBinding` in UTs to test flipping the feature gate before binding. We expect the result remains the same.